### ISSUE TITLE
Incompatible type error in regular js with JSDoc generic type annotations

### ISSUE THREAD

**Issue by zfeher** [Microsoft/TypeScript#29116]
**TypeScript Version:**  3.3.0-dev.20181221

**Search Terms:** types incompatible, jsdoc, generic

**Code**
Here I am using regular javascript code with JSDoc type annotations to add typings.

```JavaScript
/**
 * @type {(f: (p1: U) => V, g: (p1: T) => U) => (p1: T) => V}
 * @template T, U, V
 */
const compose2 = (f, g) => (p1) => f(g(p1))

/**
 * @type {(a: number) => number}
 */
const f = a => a

/**
 * @type {(a: number) => number}
 */
const g = a => a

const c = compose2(f, g)
```

**Expected behavior:** I think no error about types are incompatible should be displayed. The generic types should be replaced with the actual ones. (in contrast if I write a similar type notation for `compose2` via `d.ts` file it works flawlessly)

**Actual behavior:** a ts(2345) error:
![image](https://user-images.githubusercontent.com/2802924/50338600-0ce07700-0515-11e9-912d-e83962e7603d.png)

**Playground Link:** 

**Related Issues:**

**Comment by RyanCavanaugh** [comment]
This is just the type parameters being in the wrong place - template makes generic types, not generic functions. The correct syntax would be
```js
/**
 * @type {<T, U, V>(f: (p1: U) => V, g: (p1: T) => U) => (p1: T) => V}
 */
const compose2 = (f, g) => (p1) => f(g(p1))
```
which works as desired
